controller/work_item_labels.go

```diff
@@ -0,0 +1,74 @@
+package controller
+
+import (
+	"github.com/fabric8-services/fabric8-wit/app"
+	"github.com/fabric8-services/fabric8-wit/application"
+	"github.com/fabric8-services/fabric8-wit/jsonapi"
+	"github.com/fabric8-services/fabric8-wit/label"
+	"github.com/fabric8-services/fabric8-wit/log"
+	"github.com/fabric8-services/fabric8-wit/workitem"
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+)
+
+// WorkItemLabelsController implements the work_item_labels resource.
+type WorkItemLabelsController struct {
+	*goa.Controller
+	db     application.DB
+	config WorkItemLabelsControllerConfiguration
+}
+
+//WorkItemLabelsControllerConfiguration configuration for the WorkItemLabelsController
+type WorkItemLabelsControllerConfiguration interface {
+	GetCacheControlLabels() string
+}
+
+// NewWorkItemLabelsController creates a work_item_labels controller.
+func NewWorkItemLabelsController(service *goa.Service, db application.DB, config WorkItemLabelsControllerConfiguration) *WorkItemLabelsController {
+	return &WorkItemLabelsController{
+		Controller: service.NewController("WorkItemLabelsController"),
+		db:         db,
+		config:     config,
+	}
+}
+
+// List runs the list action.
+func (c *WorkItemLabelsController) List(ctx *app.ListWorkItemLabelsContext) error {
+	return application.Transactional(c.db, func(appl application.Application) error {
+		wi, err := appl.WorkItems().LoadByID(ctx, ctx.WiID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+		}
+		var ls []label.Label
+		labelIDs := wi.Fields[workitem.SystemLabels].([]interface{})
+		for _, lbl := range labelIDs {
+			lblStr := lbl.(string)
+			id, err := uuid.FromString(lblStr)
```

Comment:
at the design level, you can specify that `labelID` is ` `d.UUID`, which will avoid you all the manual conversion + error checking in the controller methods (see the workitem controller, for example)

Reply:
@xcoulon : here I am converting from DB wi.Fields[workitem.SystemLabels] to array of UUIDs.. so need to do manual conversion from interfaces.
right now in the migration I am using -
```
workitem.SystemLabels: {
	Type: &workitem.ListType{
	  SimpleType:    workitem.SimpleType{Kind: workitem.KindList},
	  ComponentType: workitem.SimpleType{Kind: workitem.KindLabel},
			},
}
``` 
Should I use KindUUID there in componentType ?

Reply:
ah right, my bad. I misunderstood the purpose of this block of code. However, you could look at the conversion at the `WorkItemStorage` level if you need to convert ID from `string` to `uuid.UUID` when reading/updating the DB.